assets/src/dashboard/app/views/myStories/content/storyGridItem/index.js

```diff
@@ -71,72 +67,125 @@ const StoryGridItem = ({
+
+  const memoizedStoryMenu = useMemo(
+    () => (
+      <StoryMenu
+        menuLabel={
+          isLocked
+            ? sprintf(
+                /* translators: %1$s: story title %2$s: user story is locked by.*/
+                __('Context menu for %1$s (locked by %2$s)', 'web-stories'),
+                formattedTitle,
+                story?.lockUser.name
+              )
+            : sprintf(
+                /* translators: %s: story title.*/
+                __('Context menu for %s', 'web-stories'),
+                formattedTitle
+              )
+        }
+        itemActive={isActive}
+        tabIndex={tabIndex}
+        onMoreButtonSelected={storyMenu.handleMenuToggle}
+        contextMenuId={storyMenu.contextMenuId}
+        storyId={story.id}
+        isInverted
+        menuItems={generatedMenuItems}
+        menuStyleOverrides={css`
+          /* force menu position to bottom corner */
+          margin: 0 0 0 auto;
+        `}
+      />
+    ),
+    [
+      isLocked,
+      formattedTitle,
+      isActive,
+      tabIndex,
+      storyMenu,
+      story.id,
+      story?.lockUser,
+      generatedMenuItems,
+    ]
+  );
+
   return (
     <CustomCardGridItem
       data-testid={`story-grid-item-${story.id}`}
+      onFocus={handleFocus}
+      $posterHeight={pageSize.posterHeight}
       ref={(el) => {
         itemRefs.current[story.id] = el;
       }}
-      onFocus={handleFocus}
       title={sprintf(
         /* translators: %s: story title.*/
         __('Details about %s', 'web-stories'),
-        story.title
+        formattedTitle
       )}
-      pageSize={pageSize}
     >
-      <StoryCardPreview
-        tabIndex={tabIndex}
-        storyImage={story.featuredMediaUrl}
-        storyTitle={story.title}
-        bottomAction={{
-          targetAction: story.bottomTargetAction,
-          label: bottomActionLabel,
-        }}
-      />
-      <DetailRow>
-        <CardTitle
-          tabIndex={tabIndex}
-          title={story.title}
-          titleLink={story.editStoryLink}
-          status={story?.status}
-          locked={story?.locked}
-          lockUser={story?.lockUser}
-          id={story.id}
-          displayDate={storyDate}
-          {...titleRenameProps}
-        />
-
-        <StoryMenu
-          itemActive={isActive}
-          tabIndex={tabIndex}
-          onMoreButtonSelected={storyMenu.handleMenuToggle}
-          contextMenuId={storyMenu.contextMenuId}
-          storyId={story.id}
-          menuItems={generatedMenuItems}
-        />
-      </DetailRow>
+      <Container>
+        <CardWrapper>
+          {story.featuredMediaUrl ? (
+            <Poster
+              alt={sprintf(
+                /* translators: %s: Story title. */
+                __('%s Poster image', 'web-stories'),
+                formattedTitle
+              )}
+              as="img"
+              src={story.featuredMediaUrl}
+            />
+          ) : (
+            <Poster />
+          )}
```

Comment:
💭 Nit: since we are using the same component here we could decide the props earlier and just render the same component. I don't think it really matters though, this is also readable

Reply:
good point, updated! still feels readable.